python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/ethernet-interfaces.j2

```diff
@@ -990,6 +990,127 @@ interface {{ ethernet_interface.name }}
+{%             endif %}
+   {{ tap_mac_address_cli }}
+{%         endif %}
+{# The position of "mpls_pop_all" does not match the EOS CLI to implement this "elif" with "vxlan_strip" #}
+{%         if ethernet_interface.switchport.tap.mpls_pop_all is arista.avd.defined(true) %}
+   switchport tap mpls pop all
+{%         elif ethernet_interface.switchport.tap.encapsulation.vxlan_strip is arista.avd.defined(true) %}
+   switchport tap encapsulation vxlan strip
+{%         endif %}
+{%         if ethernet_interface.switchport.tap.encapsulation.gre.strip is arista.avd.defined(true) %}
+{%             for protocol in ethernet_interface.switchport.tap.encapsulation.gre.protocols | arista.avd.natural_sort('protocol') %}
+{%                 set tap_encapsulation_cli = "switchport tap encapsulation gre protocol " ~ protocol.protocol %}
+{%                 if protocol.feature_header_length is arista.avd.defined %}
+{%                     set tap_encapsulation_cli = tap_encapsulation_cli ~ " feature header length " ~ protocol.feature_header_length ~ " " ~ "strip" %}
+{%                     if protocol.re_encapsulation_ethernet_header is arista.avd.defined(true) %}
+{%                         set tap_encapsulation_cli = tap_encapsulation_cli ~ " re-encapsulation ethernet" %}
+{%                     endif %}
+{%                 else %}
+{%                     set tap_encapsulation_cli = tap_encapsulation_cli ~ " strip" %}
+{%                 endif %}
+   {{ tap_encapsulation_cli }}
+{%             endfor %}
+{%             if ethernet_interface.switchport.tap.encapsulation.gre.protocols is not arista.avd.defined and
+                ethernet_interface.switchport.tap.encapsulation.gre.destinations is not arista.avd.defined %}
+   switchport tap encapsulation gre strip
+{%             endif %}
+{%             for destination in ethernet_interface.switchport.tap.encapsulation.gre.destinations | arista.avd.natural_sort('destination') %}
+{%                 set tap_encapsulation_cli = "switchport tap encapsulation gre destination " ~ destination.destination %}
+{%                 if destination.source is arista.avd.defined %}
+{%                     set tap_encapsulation_cli = tap_encapsulation_cli ~ " source " ~ destination.source %}
+{%                 endif %}
+{%                 for destination_protocol in destination.protocols | arista.avd.natural_sort('protocol') %}
+{%                     set tap_encapsulation_cli = tap_encapsulation_cli ~ " protocol " ~ destination_protocol.protocol %}
+{%                     if destination_protocol.feature_header_length is arista.avd.defined %}
+{%                         set tap_encapsulation_cli = tap_encapsulation_cli ~ " feature header length " ~ destination_protocol.feature_header_length ~ " " ~ "strip" %}
+{%                         if destination_protocol.re_encapsulation_ethernet_header is arista.avd.defined(true) %}
```

Comment:
same as other comment re_encapsulation_ethernet_header should not be depend on feature_header_length

Reply:
Fixed